File: backend/EnterpriseAuth.Api/Core/Application/DTOs/LocationLookupDto.cs
namespace EnterpriseAuth.Api.Core.Application.DTOs
{
    public class LocationLookupDto
    {
        public string Site { get; set; } = string.Empty;
        public string Location { get; set; } = string.Empty;
        public string Warehouse { get; set; } = string.Empty;
        public string WarehouseName { get; set; } = string.Empty;
        public string LocationType { get; set; } = string.Empty;
        public string LocationTypeName { get; set; } = string.Empty;
    }
}

File: lib/features/logistics/data/models/location_lookup_dto.dart
import '../../domain/entities/location_lookup.dart';

class LocationLookupDto {
  final String? site;
  final String? location;
  final String? warehouse;
  final String? warehouseName;
  final String? locationType;
  final String? locationTypeName;

  LocationLookupDto({
    this.site,
    this.location,
    this.warehouse,
    this.warehouseName,
    this.locationType,
    this.locationTypeName,
  });

  factory LocationLookupDto.fromJson(Map<String, dynamic> json) {
    return LocationLookupDto(
      site: json['site'],
      location: json['location'],
      warehouse: json['warehouse'],
      warehouseName: json['warehouseName'],
      locationType: json['locationType'],
      locationTypeName: json['locationTypeName'],
    );
  }

  LocationLookup toEntity() {
    return LocationLookup(
      site: site,
      location: location,
      warehouse: warehouse,
      warehouseName: warehouseName,
      locationType: locationType,
      locationTypeName: locationTypeName,
    );
  }

  Map<String, dynamic> toSqlMap() {
    return {
      'site': site,
      'location': location,
      'warehouse': warehouse,
      'warehouseName': warehouseName,
      'locationType': locationType,
      'locationTypeName': locationTypeName,
    };
  }
}
